### PR TITLE
Core Course A - outline

### PR DESCRIPTION
**Pull request by alanshaw**
Here's an attempt at a course introduction and outline

### CODE REVIEW

File: CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+Welcome space ranger to Core Course A. In this course, you can expect to:
+
+1. Learn about immutability and why it's important - isn't change the one true constant?
+1. Explore the anatomy of a CID - what's behind all those letters and numbers?
+1. Find out what the heck a Merkle DAG is - where is the Merkle forest and can I take my family?
+1. Learn what happens to your file when you add it to IPFS - where did my file go, why is it in bits and wait, my file is a tree?
+1. Understand what MFS (Mutable File System) is - what now?...you literally just told me everything was immutable!
+
+Warning, content may contain traces of the following: content vs location addressing, trust issues, IPLD format multicodecs, hash functions, multibase encoding, balanced vs trickle DAG layouts, chunking, content deduping, UnixFS.
```
Comment (alanshaw) [inline CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md:22]: Ok so, I don't know where this intro would go, whether it's on the website (or somewhere like that) or spoken as an intro actually IN the workshop. If the latter then we probably don't need these keywords, but if the former then I wanted people to be able to easily see what kind of things we'd be talking about. Like a word cloud, from the early 2000s.

File: CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+# Understanding how the InterPlanetary File System deals with Files
+
+Welcome space ranger to Core Course A. In this course, you can expect to:
+
+1. Learn about immutability and why it's important - isn't change the one true constant?
+1. Explore the anatomy of a CID - what's behind all those letters and numbers?
+1. Find out what the heck a Merkle DAG is - where is the Merkle forest and can I take my family?
+1. Learn what happens to your file when you add it to IPFS - where did my file go, why is it in bits and wait, my file is a tree?
+1. Understand what MFS (Mutable File System) is - what now?...you literally just told me everything was immutable!
```
Comment (alanshaw) [inline CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md:20]: The thing that is missing here is bitswap, but looking at the timings below we're unlikely to have time. What do you think?

File: CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+Warning, content may contain traces of the following: content vs location addressing, trust issues, IPLD format multicodecs, hash functions, multibase encoding, balanced vs trickle DAG layouts, chunking, content deduping, UnixFS.
+
+---
+
+## Part 1 - Why immutability
```
Comment (alanshaw) [inline CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md:31]: Each part _could_ be presented by a different speaker.

File: CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+Duration: 15 min
+Big wow: ???
```
Comment (alanshaw) [inline CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md:34]: I've tried to arrange these such that there's a "BIG WOW" (a visual or demo or something interactive) for each part. This is the only one that doesn't have one. I feel strongly that we should have something in this part but I don't know what yet.

File: CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+Topics:
```
Comment (alanshaw) [inline CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md:36]: I've tried to express the topics in terms of what/why/how - so we don't get lost in the details. I've tried to ask myself what others would be asking.

File: CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md (hunk 2 above)
Comment (olizilla) [reply CORE_AND_ELECTIVE_COURSES/CORE_COURSE_A/README.md:20]: Is ok. We plan to cover some of bitswap in course D (data lifecycle)

**Comment by mikeal** [comment]
LGTM, and I can present the “DAG” section if you like.

**Review by mikeal — APPROVED** [review]

**Comment by olizilla** [comment]
This might be the right course to briefly cover the different datastores (filestore, urlstore, badger) https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-filestore

**Comment by daviddias** [comment]
I'm merging this one for now with the blessing of @alanshaw :)

**Pull request merged**